### PR TITLE
Integrate llvm-project@7c69491

### PR DESCRIPTION
**Pull request by qedawkins**
Retains the revert for llvm-project@1004865

### CODE REVIEW

**Comment by qedawkins** [comment]
ok looks like I messed up the branch naming.

File: compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp [hunk 1]
```diff
@@ -56,7 +57,7 @@ LogicalResult runLLVMIRPasses(const LLVMTarget &target,
   case SanitizerKind::kAddress: {
     passBuilder.registerOptimizerLastEPCallback(
         [](llvm::ModulePassManager &modulePassManager,
-           llvm::OptimizationLevel Level) {
+           llvm::OptimizationLevel Level, llvm::ThinOrFullLTOPhase LTOPhase) {
```
Comment (kuhar) [inline compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp]: If this is unused, don't name the argument
```suggestion
           llvm::OptimizationLevel Level, llvm::ThinOrFullLTOPhase /*LTOPhase*/) {
```

File: compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp [hunk 2]
```diff
@@ -73,7 +74,7 @@ LogicalResult runLLVMIRPasses(const LLVMTarget &target,
   case SanitizerKind::kThread: {
     passBuilder.registerOptimizerLastEPCallback(
         [](llvm::ModulePassManager &modulePassManager,
-           llvm::OptimizationLevel Level) {
+           llvm::OptimizationLevel Level, llvm::ThinOrFullLTOPhase LTOPhase) {
```
Comment (kuhar) [inline compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp]: same here

**Review by kuhar — APPROVED** [review]

File: compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp (hunk 1 above)
Comment (qedawkins) [reply compiler/plugins/target/LLVMCPU/LLVMIRPasses.cpp]: I'm just going to drop the arg name, I just made it up and I don't think it adds much.

**Pull request merged**